lib/watchdog/watchdog_test.go

```diff
@@ -0,0 +1,172 @@
+package watchdog
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestWatchdog_StartStop tests that the watchdog starts a process, creates a PID
+// file, and stops the process when asked to.
+func TestWatchdog_StartStop(t *testing.T) {
+	t.Run("successful start and graceful stop", func(t *testing.T) {
+		wd := NewWatchdog("test.pid", "wd.pid", 100*time.Millisecond)
+
+		t.Cleanup(func() {
+			os.Remove("test.pid")
+			os.Remove("wd.pid")
+		})
+
+		// Mock command that will sleep for 1 second.
+		cmd := exec.Command("sleep", "1")
+		errChan := make(chan error, 1)
+
+		go func() {
+			errChan <- wd.Start(cmd.Path, cmd.Args[1:]...)
+		}()
+
+		// Wait for process to start.
+		time.Sleep(200 * time.Millisecond)
+
+		// Verify process is running.
+		wd.cmdMutex.Lock()
+		require.NotNil(t, wd.cmd)
+		require.NotNil(t, wd.cmd.Process)
+		wd.cmdMutex.Unlock()
+
+		// Stop the watchdog.
+		wd.Stop()
+
+		// Verify no errors.
+		select {
+		case err := <-errChan:
+			require.NoError(t, err)
+		case <-time.After(500 * time.Millisecond):
+			t.Fatal("timeout waiting for Start to return")
+		}
+	})
```

Comment:
Please, rewrite tests without an additional nested level.
```suggestion
// TestWatchdog_StartStop tests that the watchdog starts a process, creates a PID
// file, and stops the process when asked to.
func TestWatchdog_StartStop_successful(t *testing.T) {
	wd := NewWatchdog("test.pid", "wd.pid", 100*time.Millisecond)

	t.Cleanup(func() {
		os.Remove("test.pid")
		os.Remove("wd.pid")
	})

	// Mock command that will sleep for 1 second.
	cmd := exec.Command("sleep", "1")
	errChan := make(chan error, 1)

	go func() {
		errChan <- wd.Start(cmd.Path, cmd.Args[1:]...)
	}()

	// Wait for process to start.
	time.Sleep(200 * time.Millisecond)

	// Verify process is running.
	wd.cmdMutex.Lock()
	require.NotNil(t, wd.cmd)
	require.NotNil(t, wd.cmd.Process)
	wd.cmdMutex.Unlock()

	// Stop the watchdog.
	wd.Stop()

	// Verify no errors.
	select {
	case err := <-errChan:
		require.NoError(t, err)
	case <-time.After(500 * time.Millisecond):
		t.Fatal("timeout waiting for Start to return")
	}
}
```

Reply:
done